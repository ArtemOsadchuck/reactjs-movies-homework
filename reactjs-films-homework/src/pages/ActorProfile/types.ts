import { IPhotos } from '../../types/pages/actorProfileTypes/types';

export interface IActorPhotos {
  photos?: IPhotos[];
  photosLength?: number;
  nameAltImg: string | undefined;
}
