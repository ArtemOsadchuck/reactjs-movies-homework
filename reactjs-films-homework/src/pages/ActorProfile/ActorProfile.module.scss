$background-color: rgb(37, 37, 37);
$h2-loading-color: rgb(121, 121, 121);

.mainWrapper {
  align-items: center;
  background: $background-color;
  color: $h2-loading-color;
  margin: 0 auto;
  max-width: 1100px;
  min-height: calc(100vh - 5rem);
  padding-top: 1rem;
  position: relative;
  width: 90%;

  .photosWrapper {
    align-items: flex-end;
    background-color: rgb(93, 32, 172);
    display: flex;
    flex-direction: column;
    height: max-content;
    justify-content: flex-end;
    width: 100%;

    .photosTitle {
      text-align: start;
      width: 68%;
    }

    .photosGrid {
      width: 68%;
    }
  }

  .cardsWrapper {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
  }
}
