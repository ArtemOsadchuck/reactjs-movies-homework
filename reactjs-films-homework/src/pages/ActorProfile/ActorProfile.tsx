import React, { useState, useMemo, useEffect } from 'react';
import actorInfo from '../../mocks/actorInfo';
import actorPhotosMock from '../../mocks/actorPhotos';
import filmsWithActor from '../../mocks/filmsWithActor';

import ActorTitle from './ActorTitleInfo';
import ActorPhotos from './ActorPhotos';
import styles from './ActorProfile.module.scss';
import MovieCard from '../../components/MovieCard';
import { IMovieCard } from '../../components/MovieCard/MovieCard';
import { IPhotos } from './ActorPhotos/ActorPhotos';
import { IActorTitleProps } from './ActorTitleInfo/ActorTitleInfo';

const ActorProfile: React.FC = () => {
  const [info, setInfo] = useState<IActorTitleProps>();
  const [photos, setPhotos] = useState<Array<IPhotos>>([]);
  const [knownBy, setKnownBy] = useState<Array<IMovieCard>>([]);
  const [sortKnownBy, setSortKnownBy] = useState<Array<IMovieCard>>([]);
  const actorGridPhotosLength = 4;

  useEffect(() => {
    actorInfo.then((res) => {
      setInfo((): IActorTitleProps => res);
      return;
    });
    actorPhotosMock.then((res) => {
      setPhotos((): IPhotos[] => res.profiles);
    });
    filmsWithActor.then((res) => {
      setKnownBy((): any[] => res.cast);
    });
  });

  const sortFilmPopularity = (arr: Array<IMovieCard>) => {
    const filmsLength = 10;
    const resultFilmsLength = [...arr].sort((a, b) => {
      return a.vote_average > b.vote_average ? -1 : 1;
    });
    return resultFilmsLength.slice(0, filmsLength);
  };

  useMemo(() => {
    setSortKnownBy(sortFilmPopularity(knownBy));
  }, [knownBy]);

  return (
    <div className={styles.mainWrapper}>
      {info ? <ActorTitle props={info} /> : null}
      <ActorPhotos props={photos} photosLength={actorGridPhotosLength} />
      <div className={styles.cardsWrapper}>
        {sortKnownBy.map((el: IMovieCard) => {
          return <MovieCard key={el.id} props={el} />;
        })}
      </div>
    </div>
  );
};
export default ActorProfile;
