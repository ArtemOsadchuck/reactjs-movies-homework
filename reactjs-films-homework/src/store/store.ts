import rootStore from './rootStore/rootStore';

export type RootState = ReturnType<typeof rootStore.getState>;
export type AppDispatch = typeof rootStore.dispatch;
export default rootStore;
