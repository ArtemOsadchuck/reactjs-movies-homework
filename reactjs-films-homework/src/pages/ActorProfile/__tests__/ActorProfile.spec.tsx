import React from 'react';
import { render, cleanup } from '@testing-library/react';

import ActorProfile from '../ActorProfile';

import { Provider } from 'react-redux';
import store from '../../../store/store';

describe('ActorProfile', () => {
  let fragment: any;

  beforeEach(() => {
    const { asFragment } = render(
      <Provider store={store}>
        <ActorProfile />
      </Provider>
    );
    fragment = asFragment();
  });

  afterEach(() => {
    cleanup();
  });

  it('Snapshot', () => {
    expect(fragment).toMatchSnapshot();
  });

  it('Must have classes mainWrapper, cardsWrapper, PhotosWrapper', () => {
    expect(fragment.firstChild).toHaveClass('mainWrapper');
    expect(fragment.firstChild!.childNodes[1]).toHaveClass('cardsWrapper');
    expect(fragment.firstChild!.childNodes[0]).toHaveClass('photosWrapper');
  });
});
