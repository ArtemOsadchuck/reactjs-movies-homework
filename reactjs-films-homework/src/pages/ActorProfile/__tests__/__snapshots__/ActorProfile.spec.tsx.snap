// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActorProfile Snapshot 1`] = `
<DocumentFragment>
  <div
    class="mainWrapper"
  >
    <div
      class="photosWrapper"
    >
      <div
        class="photosGrid"
      />
    </div>
    <div
      class="cardsWrapper"
    />
  </div>
</DocumentFragment>
`;
